### PR TITLE
aubio: fix build on linux

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

#93940

### CODE REVIEW

File: Formula/aubio.rb [hunk 1]
```diff
@@ -25,20 +25,38 @@ class Aubio < Formula
   depends_on "numpy"
   depends_on "python@3.9"
 
+  # Replace python reference with python3
+  patch :DATA
```
Comment (carlocab) [inline Formula/aubio.rb:29]: Maybe we can just do
```ruby
ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
```
?

Comment (branchvincent) [reply Formula/aubio.rb:29]: I wanted to do that too but it seems the formula doesn't have a `bin/python`:

```
$ ls $(brew --prefix python@3.9)/bin
2to3  2to3-3.9  idle3  idle3.9  pip3  pip3.9  pydoc3  pydoc3.9  python3  python3-config  python3.9  python3.9-config  wheel3
```

Comment (danielnachun) [reply Formula/aubio.rb:29]: I'm not entirely sure if that will work as it seems that a particular script has a hard coded reference to a `python` executable that doesn't exist in the `python@3.9` bottle.

Comment (carlocab) [reply Formula/aubio.rb:29]: It does in `libexec`:
```
❯ ls $(brew --prefix python@3.9)/libexec/bin
idle  pip  pydoc  python  python-config  wheel
```

Comment (branchvincent) [reply Formula/aubio.rb:29]: ah, perfect!

**Comment by cho-m** [comment]
Test failure probably due to needing alternative Linux dependencies to replace Apple CoreAudio if we want to test Apple AIFF format (and to enable features for more media formats).

See https://github.com/aubio/aubio/blob/master/python/README.md#built-with
I guess something like `ffmpeg` or `libsndfile` should add support.

```
==> /home/linuxbrew/.linuxbrew/Cellar/aubio/0.4.9_2/bin/aubiocut --verbose wood24.aiff
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/aubio/0.4.9_2/libexec/bin/aubiocut", line 33, in <module>
    sys.exit(load_entry_point('aubio==0.4.9', 'console_scripts', 'aubiocut')())
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/aubio/cut.py", line 147, in main
    timestamps, total_frames = _cut_analyze(options)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/aubio/cut.py", line 102, in _cut_analyze
    s = source(source_uri, samplerate, hopsize)
RuntimeError: AUBIO ERROR: source_wavread: Failed opening wood24.aiff (could not find RIFF header)
Error: aubio: failed
```

File: Formula/aubio.rb [hunk 2]
```diff
@@ -25,20 +25,29 @@ class Aubio < Formula
...
+    # Ensure `python` references use our python3
+    ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
+
+    system "python3", "./waf", "configure", "--prefix=#{prefix}"
+    system "python3", "./waf", "build"
+    system "python3", "./waf", "install"
 
-    system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(prefix)
+    system "python3", *Language::Python.setup_install_args(prefix)
     bin.env_script_all_files(libexec/"bin", PYTHONPATH: ENV["PYTHONPATH"])
```
Comment (carlocab) [inline Formula/aubio.rb:45]:
```suggestion
```
This isn't needed. `PYTHONPATH` is not set here, so we're not really pointing this to anywhere.

**Review by carlocab — APPROVED** [review]

**Comment by cho-m** [comment]
Can merge for Python fixes, but would still need a follow up PR to get Linux to bottle.

Mainly should find some alternatives for macOS defaults to provide feature parity (https://github.com/aubio/aubio/blob/master/doc/requirements.rst#platform-notes):
- macOS uses Accelerate to compute FFT. Not sure how Linux default compares, or if worth adding `fftw`.
- macOS uses CoreAudio/AudioToolbox for decoding audio. Linux default only supports WAV format, so probably should add some dependencies (need to see which among `libav`, `libsamplerate`, and `libsndfile` provides reasonable feature set to match our macOS distribution).

---

ArchLinux for comparison https://archlinux.org/packages/community/x86_64/aubio/ (though may have more features than we enable on macOS build):
- `fftw`
- `ffmpeg@4`
- `jack`
- `libsamplerate`
- `libsndfile`

**Review by carlocab — APPROVED** [review]
Let's merge this. Adding new deps can be done in a separate PR.

**Pull request merged**